src/Compilers/Core/Portable/MetadataReader/MetadataTypeName.cs

```diff
@@ -68,6 +119,8 @@ internal partial struct MetadataTypeName
         /// </summary>
         private ImmutableArray<string> _namespaceSegments;
 
+        private ImmutableArray<ReadOnlyMemory<char>> _namespaceSegmentsMemory;
```

Comment:
it's a pity this is still an IA (which will allocate the underlying array).  Esp. as most cases will be splitting a string with <4 actual namespace segments.  Having some slim array-like type that did not hit the heap for the low-number-of-elements case would be ideal here.

Reply:
I thought that was 'TemporaryArray'. I guess we don't want to use that type as a field?